tests/trainer/test_checkpoint.py

```diff
@@ -279,12 +280,8 @@ def test_checkpoint(
         composer_trainer_hparams.model = model_hparams.model
         composer_trainer_hparams.optimizer = model_hparams.optimizer
         composer_trainer_hparams.schedulers = model_hparams.schedulers
-    if not isinstance(composer_trainer_hparams.train_dataset, SyntheticHparamsMixin):
-        pytest.skip("Checkpointing tests require synthetic data")
-        return
-    if not isinstance(composer_trainer_hparams.val_dataset, SyntheticHparamsMixin):
-        pytest.skip("Checkpointing tests require synthetic data")
-        return
+        # to ensure that synthetic data is properly cast
+        composer_trainer_hparams.dataloader.pin_memory = True
```

Comment:
In the tests we configure 0 dataloader works, which requires pin_memory to be false. Curious why it needs to be True? I don't think memory pinning should be doing any casting?

Reply:
Cool, I think I was just able to remove this just Austin got rid of the `batch_to_device` function that Austin talked about offline!